feg/gateway/policydb/flow_parser.go

```diff
@@ -84,17 +85,20 @@ func getFlowDescriptionFromSplit(matches []string) (*protos.FlowDescription, err
 		Match: &protos.FlowMatch{
 			Direction: direction,
 			IpProto:   proto,
-			IpSrc: &protos.IPAddress{
-				Version: protos.IPAddress_IPV4,
-				Address: []byte(src.ip),
-			},
-			IpDst: &protos.IPAddress{
-				Version: protos.IPAddress_IPV4,
-				Address: []byte(dst.ip),
-			},
 		},
 	}
-
+	if src.ip != "" {
+		rule.Match.IpSrc = &protos.IPAddress{
+			Version: src.version,
+			Address: []byte(src.ip),
+		}
+	}
+	if dst.ip != "" {
+		rule.Match.IpDst = &protos.IPAddress{
+			Version: dst.version,
+			Address: []byte(dst.ip),
+		}
+	}
```

Comment:
On the previous lines I see `matches[0]`... `matches[4]`. Shouldn't you check len of matches to avoid accessing an out of bound index?

```
func getFlowDescriptionFromSplit(matches []string)
	if (len(matches) != 4) {
		return nil, fmt.Error("Wrong matcher string. Number of matches different than 4")	
}
```